File: WebApp/indoor-localization/src/data-access/mock-repositories/MockZoneRepository.ts
import { Zone } from "../../entities/Zone";
import { IZoneRepository } from "../repository-interfaces/IZoneRepository";

export class MockZoneRepository implements IZoneRepository {
    private static zones: Zone[] = [];

    async Get(id: number): Promise<Zone | null> {
        return MockZoneRepository.zones.find(zone => zone.id === id) || null;
    }

    async GetAll(): Promise<Zone[]> {
        return MockZoneRepository.zones;
    }

    async Add(zone: Zone): Promise<boolean> {
        MockZoneRepository.zones.push(zone);
        return true;
    }

    async Delete(id: number): Promise<boolean> {
        const index = MockZoneRepository.zones.findIndex(zone => zone.id === id);
        if (index !== -1) {
            MockZoneRepository.zones.splice(index, 1);
            return true;
        }
        return false;
    }

    async Update(updatedZone: Zone): Promise<boolean> {
        const index = MockZoneRepository.zones.findIndex(zone => zone.id === updatedZone.id);
        if (index !== -1) {
            MockZoneRepository.zones[index] = updatedZone;
            return true;
        }
        return false;
    }
}

File: Android/indoor_localization/lib/domain/mock-repositories/mock_zone_repository.dart
import 'package:indoor_localization/domain/entities/zone.dart';
import 'package:indoor_localization/domain/repository-interfaces/zone_repository.dart';

class MockZoneRepository implements IZoneRepository {
  static final List<Zone> _zones = [];

  @override
  Future<Zone> get(int id) async {
    return _zones.firstWhere((zone) => zone.id == id);
  }

  @override
  Future<List<Zone>> getAll() async {
    return _zones;
  }

  @override
  bool add(Zone zone) {
    _zones.add(zone);
    return true;
  }

  @override
  bool delete(int id) {
    final index = _zones.indexWhere((zone) => zone.id == id);
    if (index != -1) {
      _zones.removeAt(index);
      return true;
    }
    return false;
  }

  @override
  bool update(Zone updatedZone) {
    final index = _zones.indexWhere((zone) => zone.id == updatedZone.id);
    if (index != -1) {
      _zones[index] = updatedZone;
      return true;
    }
    return false;
  }
}


